libs/api/domains/rights-portal/src/lib/api-domains-rights-portal.service.ts

```diff
@@ -0,0 +1,66 @@
+import { LOGGER_PROVIDER } from '@island.is/logging'
+import type { Logger } from '@island.is/logging'
+import { Inject, Injectable } from '@nestjs/common'
+import {
+  AidsAndNutritionDTO,
+  AidsandnutritionApi,
+  TherapyApi,
+  TherapyDTO,
+} from '@island.is/clients/icelandic-health-insurance/rights-portal'
+import { ApolloError } from 'apollo-server-express'
+import { Auth, AuthMiddleware, User } from '@island.is/auth-nest-tools'
+
+/** Category to attach each log message to */
+const LOG_CATEGORY = 'rights-portal-service'
+
+@Injectable()
+export class RightsPortalService {
+  constructor(
+    @Inject(LOGGER_PROVIDER) private readonly logger: Logger,
+    private therapyApi: TherapyApi,
+    private aidsAndNutritionApi: AidsandnutritionApi,
+  ) {}
+
+  handleError(error: any, detail?: string): ApolloError | null {
+    this.logger.error(detail || 'Rights portal error', {
+      error: JSON.stringify(error),
+      category: LOG_CATEGORY,
+    })
+    throw new ApolloError('Failed to resolve request', error.status)
+  }
+
+  private handle4xx(error: any, detail?: string): ApolloError | null {
+    if (error.status === 403 || error.status === 404) {
+      return null
+    }
+    return this.handleError(error, detail)
+  }
```

Comment:
Maybe you copied this from elsewhere but we should start moving away from this error handling pattern. Enhanced Fetch includes standardized error logging, and unexpected errors should not be ignored or wrapped with ApolloError.

Also, the return type can be improved to simplify type handling elsewhere. `handleError` always throws so returns `never` and `handle4xx` only returns `null` (or throws an error).

Reply:
Yeah, simplified